lib/Dialect/FIRRTL/Transforms/DropConst.cpp

```diff
@@ -0,0 +1,184 @@
+//===- DropConst.cpp - Check and remove const types -------------*- C++ -*-===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+//
+// This file defines the DropConst pass.
+//
+//===----------------------------------------------------------------------===//
+
+#include "PassDetails.h"
+#include "circt/Dialect/FIRRTL/FIRRTLOps.h"
+#include "circt/Dialect/FIRRTL/FIRRTLTypes.h"
+#include "circt/Dialect/FIRRTL/FIRRTLUtils.h"
+#include "circt/Dialect/FIRRTL/FIRRTLVisitors.h"
+#include "circt/Dialect/FIRRTL/Passes.h"
+
+#define DEBUG_TYPE "drop-const"
+
+using namespace circt;
+using namespace firrtl;
+
+namespace {
+
+class OpVisitor : public FIRRTLVisitor<OpVisitor> {
+public:
+  using FIRRTLVisitor<OpVisitor>::visitExpr;
+  using FIRRTLVisitor<OpVisitor>::visitDecl;
+  using FIRRTLVisitor<OpVisitor>::visitStmt;
+
+  LogicalResult checkModule(FModuleLike module) {
```

Comment:
"check" implies to me that it is not modifying the IR.

Reply:
Renamed.